simulator/simulator.go

```diff
@@ -0,0 +1,124 @@
+package simulator
+
+import (
+	"errors"
+	"sync"
+)
+
+// FakeHost is the interface which defines the pluggable behavior of a simulated
+// host. Each "type" of host may have their own implementation of this interface.
+type FakeHost interface {
```

Comment:
I'm curious as to the strategy here. I would think that the Host interface would have the same methods as the osquery launcher. It needs to exercise all endpoints that the launcher does(enroll, log, config, read, write).